src/main/java/com/openlattice/datastore/data/controllers/DataController.java

```diff
@@ -1143,6 +1062,44 @@ private static OffsetDateTime getDateTimeFromLong( long epochTime ) {
         return OffsetDateTime.ofInstant( Instant.ofEpochMilli( epochTime ), ZoneId.systemDefault() );
     }
 
+    private WriteEvent clearOrDeleteEntities( UUID entitySetId, Set<UUID> entityKeyIds, DeleteType deleteType ) {
+        int numUpdates = 0;
+        long maxVersion = Long.MIN_VALUE;
+
+        // access checks for entity set and properties
+        final Map<UUID, PropertyType> authorizedPropertyTypes =
+                getAuthorizedPropertyTypesForDelete( entitySetId, Optional.empty(), deleteType );
+
+        Iterable<List<UUID>> entityKeyIdChunks = Iterables.partition( entityKeyIds, MAX_BATCH_SIZE );
+        for ( List<UUID> chunkList : entityKeyIdChunks ) {
+            Set<UUID> chunk = Sets.newHashSet( chunkList );
+
+            WriteEvent writeEvent;
+
+            if ( deleteType == DeleteType.Hard ) {
+                deleteAssociations( entitySetId, Optional.of( chunk ) );
+                writeEvent = dgm.deleteEntities(
+                        entitySetId,
+                        chunk,
+                        authorizedPropertyTypes );
+            } else {
+                List<WriteEvent> clearEvents = clearAssociations( entitySetId, Optional.of( chunk ) );
+                while ( !clearEvents.isEmpty() ) {
+                    clearEvents = clearAssociations( entitySetId, Optional.of( chunk ) );
```

Comment:
This looks identical to the one above. How does `clearAssociations` work?

Reply:
Ooh thanks good catch, this was left over from a previous iteration